src/functional/iterator/embedded.py

```diff
@@ -143,8 +142,33 @@ def __array__(self) -> np.ndarray:
         ...
 
 
-def _is_column(v: Any) -> TypeGuard[Column]:
-    return isinstance(v, np.ndarray)
+class Column(numpy.lib.mixins.NDArrayOperatorsMixin):
+    def __init__(self, kstart: int, data: np.ndarray):
+        self.kstart = kstart
+        self.data = data
+
+    def __getitem__(self, i: int):
+        return self.data[i - self.kstart]
+
+    def tuple_get(self, i: int):
+        if self.data.dtype.names is not None:
+            return Column(self.kstart, self.data[f"f{i}"])
```

Comment:
```suggestion
        if self.data.dtype.names:
            return Column(self.kstart, self.data[self.data.dtype.names[i]])
```
Seen this cleaner version in one of Felix's PRs.